### ISSUE TITLE
ACCESSING LEFT AND RIGHT EDGES WITH UNITS IN ARBITRARY_GRID

### ISSUE THREAD

**Issue by samlelouch** [yt-project/yt#2087]
### Bug report

**Bug summary**
units supplied to `left_edge` and `right_edge` of `arbitrary_grid` are not being recognized unless supplied using `ds.arr`

```
dd=ds.arbitrary_grid(left_edge=[(com_x_1.value-30), (com_y_1.value-30), (com_z_1.value-30)]*kpc, right_edge=[(com_x_1.value+30), (com_y_1.value+30), (com_z_1.value+30)]*kpc, dims=[128, 128, 128])
print(dd['gas', 'velocity_x'])
```
ERROR:
>   yt.units.unit_registry.UnitParseError: Code units have not been defined.  Try creating the array or quantity using ds.arr or ds.quan instead.

Using,
`left_edge= ds.arr([(com_x_1.value-30), (com_y_1.value-30), (com_z_1.value-30)], 'kpc')`
the issue doesn't come up.

**Version Information**

  * Operating System: Ubuntu
  * Python Version: 3.6
  * yt version: 3.4 installed from conda
  * Other Libraries (if applicable):

**Comment by ngoldbaum** [comment]
Thanks for the detailed report!

**Comment by munkm** [comment]
Fixed by #2119
